grype/db/internal/gormadapter/open.go

```diff
@@ -16,6 +16,13 @@ var writerStatements = []string{
 	`PRAGMA journal_mode = MEMORY`,
 }
 
+var readOptions = []string{
+	"&immutable=1",
+	"&cache=shared",
+	"&mode=ro",
+	"&_journal_mode=WAL",
```

Comment:
I agree with the conversation out in comments that `WAL` shouldn't help us out from a readers perspective --we should probably remove this option.